a3p-integration/proposals/z:acceptance/test-lib/kread.js

```diff
@@ -0,0 +1,357 @@
+};
+
+export const getMarketItemsChildren = async () => {
+  const { children } = await agd.query([
+    'vstorage',
+    'children',
+    `published.kread.market-items`,
+  ]);
+
+  return children;
+};
+
+export const getMarketItem = async itemNode => {
+  const itemMarketPath = `:published.kread.market-items.${itemNode}`;
+  const rawItemData = await agoric.follow('-lF', itemMarketPath, '-o', 'text');
+  const item = marshaller.fromCapData(JSON.parse(rawItemData));
+
+  return item;
+};
+
+export const getCharacterInventory = async characterName => {
+  const inventoryPath = `kread.character.inventory-${characterName}`;
+  const characterInventory = await getContractInfo(inventoryPath, {
+    agoric,
+    prefix: 'published.',
+  });
+
+  return characterInventory;
+};
+
+export const getBalanceFromPurse = async (address, type) => {
+  const walletRaw = await agoric.follow(
+    '-lF',
+    `:published.wallet.${address}.current`,
+    '-o',
+    'text',
+  );
+  const purses = marshaller.fromCapData(JSON.parse(walletRaw)).purses;
+
+  const assetBrands = {
+    character: brands.KREAdCHARACTER,
+    item: brands.KREAdITEM,
+  };
+  const assetBrand = assetBrands[type];
+  if (!assetBrand) {
+    throw new Error('Invalid type provided. Must be "character" or "item".');
+  }
+
+  const assetPurse = purses.find(
+    ({ brand }) => brand.getBoardId() === assetBrand.getBoardId(),
+  );
+
+  return assetPurse?.balance.value.payload[0]?.[0] || null;
+};
+
+/** @import {Brand, CopyBagAmount} from '@agoric/ertp/src/types.js'; */
+
+/**
+ * @param {Brand} brand
+ * @param {any} asset
+ * @returns {CopyBagAmount<'item'>}
+ */
+export const assetAsAmount = (brand, asset) => {
+  const assetValue = makeCopyBag([[asset, 1n]]);
+  const assetAmount = AmountMath.make(brand, assetValue);
+  return assetAmount;
+};
+
+export const totalAssetPriceAmount = asset => {
+  const fees = AmountMath.add(asset.platformFee, asset.royalty);
+  const price = AmountMath.add(asset.askingPrice, fees);
+  return price;
+};
+const mintCharacterOffer = async () => {
+  const id = `KREAd-mint-character-acceptance-test`;
+  const body = {
+    method: 'executeOffer',
+    offer: {
+      id,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: ['kread'],
+        callPipe: [['makeMintCharacterInvitation', []]],
+      },
+      offerArgs: { name: 'ephemeral_Ace' },
+      proposal: {
+        give: {
+          Price: {
+            brand: brands.IST,
+            value: 5n * ISTunit,
+          },
+        },
+      },
+    },
+  };
+
+  return JSON.stringify(marshaller.toCapData(harden(body)));
+};
+
+const unequipAllItemsOffer = async address => {
+  const kreadCharacter = await getBalanceFromPurse(address, 'character');
+  if (!kreadCharacter) {
+    throw new Error('Character not found on user purse');
+  }
+
+  const inventoryKeyId = kreadCharacter.keyId === 1 ? 2 : 1;
+  const kreadCharacter2 = { ...kreadCharacter, keyId: inventoryKeyId };
+
```

Comment:
This function seems to be doing something other than `unequipAllItemsOffer`. I have no problem with a single purpose function for a test, but it shouldn't be named as if it were reusable. 
If I'm misunderstanding what's happening with character 2, please add a comment to clarify.

Reply:
The way KREAd manages the character items inventory involves minting two identical versions of the same character: one is stored in the user's wallet, and the other is bound to an inventory seat, along with any equipped items.
The only difference between these two character is their keyId, which can either be 1 or 2.

When an offer is created to unequip all items, it's necessary to define the proposal attribute "give" as the character held in the user's wallet and "want" as the character currently occupying the inventory seat.

 This line of code
 `const inventoryKeyId = kreadCharacter.keyId === 1 ? 2 : 1;`
 , ensures that you correctly identify which character is held in the user's wallet, determining which character to reference for "want" and "give."

Reply:
> one is stored in the user's wallet, and the other is bound to an inventory seat, along with any equipped items.

Oh, yes. I do remember learning that more than once.

Please add a comment, because I'm going to forget again. I think I'd say something like 
```
// deal with KREAd's funky parallel character representation swap
```
Maybe this will be enough to jog my memory.

Reply:
Change request addressed